### PR TITLE
Send a pod event if preStop hook did not complete in time.

### PR DESCRIPTION
**Pull request by aveshagarwal**
Fixes https://github.com/kubernetes/kubernetes/issues/31902
@kubernetes/rh-cluster-infra @derekwaynecarr

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31911)

### CODE REVIEW

File: pkg/kubelet/events/event.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ const (
 	FailedValidation = "FailedValidation"
 
 	// Lifecycle hooks
-	FailedPostStartHook = "FailedPostStartHook"
-	FailedPreStopHook   = "FailedPreStopHook"
+	FailedPostStartHook    = "FailedPostStartHook"
+	FailedPreStopHook      = "FailedPreStopHook"
+	UncompletedPreStopHook = "UncompletedPreStopHook"
```
Comment (ncdc) [inline pkg/kubelet/events/event.go]: Incomplete or TimedOut or something like that, maybe?

File: pkg/kubelet/dockertools/docker_manager.go [hunk 2]
```diff
@@ -1506,6 +1506,8 @@ func (dm *DockerManager) killContainer(containerID kubecontainer.ContainerID, co
 		select {
 		case <-time.After(time.Duration(gracePeriod) * time.Second):
 			glog.V(2).Infof("preStop hook for container %q did not complete in %d seconds", name, gracePeriod)
+			message := fmt.Sprintf("preStop hook for container %q did not complete in %d seconds", name, gracePeriod)
```
Comment (ingvagabund) [inline pkg/kubelet/dockertools/docker_manager.go]: What does it mean for a hook to complete? Is it the common terminology? What about "did not finish"?

File: pkg/kubelet/events/event.go (hunk 1 above)
Comment (ingvagabund) [inline pkg/kubelet/events/event.go]: Unfinished?

**Comment by aveshagarwal** [comment]
Rebased. @derekwaynecarr PTAL.

**Comment by derekwaynecarr** [comment]
LGTM

**Pull request merged**